helpers/parse_helpers.go

```diff
@@ -207,14 +207,13 @@ func CreateDecryptCryptoConfig(keys []string, decRecipients []string) (encconfig
 		return encconfig.CryptoConfig{}, err
 	}
 
-	if len(x509s) > 0 {
```

Comment:
oh my bad, I thought x509 certs are mandatorily  required for protocols other than keyprovider.  But the fix could fail for key provider as we wont be passing any files, and this function would through up "no such file or directory" error.
Could we add a check in this function `processx509Certs `  read file only if it exists, if its not a file doesnt exists then return nil, nil  

`func processx509Certs(keys []string) ([][]byte, error) {
	var x509s [][]byte
	for _, key := range keys {
                if _, err := os.Stat(strings.Split(key, ":")[0]); err == nil {
		tmp, err := ioutil.ReadFile(strings.Split(key, ":")[0])
		if err != nil {
			return nil, errors.Wrap(err, "Unable to read file")
		}
            
		if !encutils.IsCertificate(tmp) {
			continue
		}
		x509s = append(x509s, tmp)
            }
	}
	return x509s, nil
}`

Reply:
I see yea we can do something like that.

Reply:
I just pushed a commit - can you verify on your side that this will work with keyprovider?

Reply:
Oh I already approved. Fix looks good to me, you can go ahead and merge. I will validate tomorrow.